ts/packages/ts/src/commands/EvalCommand.ts

```diff
@@ -127,6 +127,28 @@ export class EvalCommand extends Command<EvalCommandReturn> {
     });
   }
 
+  private shouldOverrideOrigin(nextOrigin: EvaluatedConfigOrigin, previousOrigin?: EvaluatedConfigOrigin): boolean {
+    switch (previousOrigin) {
+      case undefined:
+        return true;
+      case EvaluatedConfigOrigin.EmptyValue:
+        return nextOrigin !== EvaluatedConfigOrigin.EmptyValue;
```

Comment:
Considering
- `const mergedIsEmpty = !mergedResult || mergedResult.result.origin === EvaluatedConfigOrigin.EmptyValue;`
- `if (mergedIsEmpty || shouldOverrideMerged) {`
This test is either irrelevant or not doing it's job.. if the previous is EmptyValue the override will happen no matter the result from here

Also, NIT: why not combine both tests to one helper method ?